### PR TITLE
[ACA-3315] - add predefined extension guard

### PR DESCRIPTION
**Pull request by SilviuCPopa**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
https://issues.alfresco.com/jira/browse/ACA-3315

**What is the new behaviour?**
Added new predefined auth guard

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: projects/aca-shared/src/lib/adf-extensions/extensions-data-loader.guard.ts [hunk 1]
```diff
@@ -75,4 +75,8 @@ export class ExtensionsDataLoaderGuard implements CanActivate {
       })
     );
   }
+
+  canActivateChild(route: ActivatedRouteSnapshot): Observable<boolean> {
```
Comment (popovicsandras) [inline projects/aca-shared/src/lib/adf-extensions/extensions-data-loader.guard.ts:79]: I'm not sure it is needed. Wouldn't this result to load always on every child route? We want to load these loaders only once, not on every child routes.

**Comment by SilviuCPopa** [comment]
> Test missing

I don't think it's needed since `canActivateChild` calls only the `canActivate` function which is already tested

File: projects/aca-shared/src/lib/adf-extensions/extensions-data-loader.guard.ts [hunk 2]
```diff
@@ -45,34 +45,46 @@ export const EXTENSION_DATA_LOADERS = new InjectionToken<
 
 @Injectable({ providedIn: 'root' })
 export class ExtensionsDataLoaderGuard implements CanActivate {
+  involved = false;
```
Comment (popovicsandras) [inline projects/aca-shared/src/lib/adf-extensions/extensions-data-loader.guard.ts:48]: might be better to call it invoked? 🤔

Comment (popovicsandras) [reply projects/aca-shared/src/lib/adf-extensions/extensions-data-loader.guard.ts:48]: And it is supposed to be private. Only this class can modify it.

File: projects/aca-shared/src/lib/adf-extensions/extensions-data-loader.guard.spec.ts [hunk 3]
```diff
@@ -131,5 +131,32 @@ describe('ExtensionsDataLoaderGuard', () => {
...
+      const extensionLoaders = {
+        fct1: function() {
+          return subject1.asObservable();
+        }
+      };
+      const extensionLoaderSpy = spyOn(extensionLoaders, 'fct1');
+      const guard = new ExtensionsDataLoaderGuard([extensionLoaders.fct1]);
+
+      subject1.next();
+      expect(guard.involved).toEqual(false);
```
Comment (popovicsandras) [inline projects/aca-shared/src/lib/adf-extensions/extensions-data-loader.guard.spec.ts:146]: This needs to be private. Unit test needs to test the behaviour of the function. This way, you are doing nothing else, but testing the implementation, which is something which we should always avoid.

File: projects/aca-shared/src/lib/adf-extensions/extensions-data-loader.guard.spec.ts [hunk 4]
```diff
@@ -131,5 +131,32 @@ describe('ExtensionsDataLoaderGuard', () => {
...
+      guard.canActivate(route).subscribe(emittedSpy, erroredSpy, completedSpy);
+      expect(guard.involved).toEqual(true);
+      expect(emittedSpy).toHaveBeenCalledWith(true);
+      expect(erroredSpy).not.toHaveBeenCalled();
+      expect(completedSpy).toHaveBeenCalled();
+      expect(extensionLoaderSpy).toHaveBeenCalled();
```
Comment (popovicsandras) [inline projects/aca-shared/src/lib/adf-extensions/extensions-data-loader.guard.spec.ts:152]: It is enough to test if this callback has been called or not. This is the behaviour what we need to test.

**Pull request merged**